### PR TITLE
gitolite: wrap gitolite-shell

### PR DESCRIPTION
**Pull request by bbigras**
git wasn't found when used with services.fcgiwrap
for http auth with nginx

###### Motivation for this change
I was trying to use gitolite with fcgiwrap and gitolite-shell woudln't find the git binary.

I thought of using `substituteInPlace` at first but there's a lot of "git" occurences that shouldn't be replaced in that script.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

/nix/store/mnf5qhaxyqvzmb5bp7h6zv03vhf3brh0-gitolite-3.6.11	  81.6M
/nix/store/m1sg02d5yvazcf5hs4hxzpfybm209gnq-gitolite-3.6.11	 226.4M

Is the size increase normal?

###### Notify maintainers

cc @thoughtpolice @lassulus @tomberek

### CODE REVIEW

**Review by tomberek — COMMENTED** [review]
I could be out of date on the current best practice: 

git can go in propagatedBuildInputs and makeWrapper can go in nativeBuildInputs

**Comment by bbigras** [comment]
@tomberek I did the changes and I added a test. It's my first one.

File: nixos/tests/gitolite-fcgiwrap.nix [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                };
+
+                services.nginx = {
+                  enable = true;
+                  recommendedProxySettings = true;
+                  virtualHosts."server".locations."/git".extraConfig = ''
+                    # turn off gzip as git objects are already well compressed
+                    gzip off;
+
+                    # use file based basic authentication
```
Comment (bbigras) [inline nixos/tests/gitolite-fcgiwrap.nix]: btw those comments were copy/pasted from some gist. I hope it's still fine.

**Review by tomberek — APPROVED** [review]
Thank you for the nixos test! This is awesome.

Builds and tests and functions fine. [NixOS]

**Review by Lassulus — APPROVED** [review]
tested with nix-review, ran the tests, LGTM

**Pull request merged**